src/react-components/media-browser.js

```diff
@@ -129,6 +129,7 @@ class MediaBrowser extends Component {
     const urlSource = this.getUrlSource(searchParams);
     newState.showNav = !!(searchParams.get("media_nav") !== "false");
     newState.selectAction = searchParams.get("selectAction") || "spawn";
+    newState.query = searchParams.get("q") || "";
```

Comment:
it looks like this should be reflected on line 128. do we want to discard the value in `state` here in all cases? i haven't looked at this code in a while

Reply:
From what I can tell `getStoreAndHistoryState` is called only on page load and whenever the store has been updated, in both cases it seems like we want to overwrite the state. I will combine this with the line above though.

Reply:
From what I can tell `getStoreAndHistoryState` is called only on page load and whenever the store has been updated, in both cases it seems like we want to overwrite the state. I will combine this with the line above though.